doc/cookbook/source/examples/gaussian_processes/gaussian_process_regression.rst

```diff
@@ -0,0 +1,59 @@
+===========================
+Gaussian Process Regression
+===========================
+
+A Gaussian Process is the extension of the Gaussian distribution to infinite dimensions. It is specified by a mean function :math:`m(\mathbf{x})` and a covariance kernel :math:`k(\mathbf{x},\mathbf{x}')` (where :math:`\mathbf{x}\in\mathcal{X}` for some input domain :math:`\mathcal{X}`). They define a distribution over real valued functions :math:`f(\cdot)`:
+
+.. math::
+	f(\mathbf{x})\sim\mathcal{GP}(m(\mathbf{x}), k(\mathbf{x},\mathbf{x}))
+
+Any finite set of such function values :math:`\mathbf{f}` is jointly Gaussian with mean specified by the mean function evaluated at the inputs :math:`\mathbf{x}_i` and covariance matrix :math:`\mathbf{C}` with entries :math:`\mathbf{C}_{ij}=k(\mathbf{x}_i,\mathbf{x}_j)`.
+
+In our regression model, we take our target variable :math:`\mathbf{y}` to be our (latent) function values :math:`\mathbf{f}` plus Gaussian noise:
+
+.. math::
+	p(\mathbf{y}|\mathbf{f})=\mathcal{N}(\mathbf{f},\sigma^2\mathbf{I})
+
+See :cite:`Rasmussen2005GPM` for a comprehensive treatment of Gaussian Processes.
+
+-------
+Example
+-------
+
+Imagine we have files with training and test data. We create `CDenseFeatures` (here 64 bit floats aka RealFeatures) and :sgclass:`CRegressionLabels` as:
+
+.. sgexample:: gaussian_process_regression.sg:create_features
+
+To fit the input (training) data :math:`\mathbf{X}`, we have to choose an appropriate :sgclass:`CMeanFunction` and  :sgclass:`CKernel` and instantiate it. Here we use a basic :sgclass:`CZeroMean` and a :sgclass:`CGaussianKernel` with width 1.
```

Comment:
Please dont explicitly mention the chosen parameters in the text. If they change for some reason, the text gets outdated. The code speaks for itself -- just mention the parameters are set